components/Tag/index.js

```diff
@@ -1,23 +1,23 @@
 import { useState } from "react";
 import Image from "next/image";
+import { StyledTagButton } from "../StyledTagButton/StyledTagButton";
 
 export default function Tag({ tag, foundTag, editTag, activityId, tagId }) {
-  const [opacity, setOpacity] = useState(foundTag ? 1 : 0.5);
+  const [opacity, setOpacity] = useState(foundTag ? 1 : 0.3);
 
   function handleClick(event) {
     editTag(activityId, tagId);
-    setOpacity(opacity === 1 ? 0.5 : 1);
+    setOpacity(opacity === 1 ? 0.3 : 1);
   }
-
   return (
-    <button onClick={handleClick} key={tag.id}>
+    <StyledTagButton onClick={handleClick} key={tag.id}>
```

Comment:
try to be explicit about the types:

```suggestion
    <StyledTagButton onClick={handleClick} key={tag.id} type="button">
```